ui/src/notebooks/pipes/Visualization/view.tsx

```diff
@@ -2,19 +2,93 @@ import React, {FC, useContext} from 'react'
 import {PipeProp} from 'src/notebooks'
 import EmptyQueryView, {ErrorFormat} from 'src/shared/components/EmptyQueryView'
 import ViewSwitcher from 'src/shared/components/ViewSwitcher'
+import {ViewTypeDropdown} from 'src/timeMachine/components/view_options/ViewTypeDropdown'
 import {checkResultsLength} from 'src/shared/utils/vis'
+import {ViewType} from 'src/types'
+import {createView} from 'src/views/helpers'
 
 // NOTE we dont want any pipe component to be directly dependent
 // to any notebook concepts as this'll limit future reusability
 // but timezone seems like an app setting, and its existance within
 // the notebook folder is purely a convenience
 import {AppSettingContext} from 'src/notebooks/context/app'
 
-const Visualization: FC<PipeProp> = ({data, results, Context, loading}) => {
+const Visualization: FC<PipeProp> = ({
+  data,
+  results,
+  onUpdate,
+  Context,
+  loading,
+}) => {
   const {timeZone} = useContext(AppSettingContext)
 
+  const updateType = (type: ViewType) => {
```

Comment:
I will 100% concede that this entire function is garbage. converting between types like this is a recipe for bugs by creating a "m:n" relationship that you have to track all the permutations of. but any changes would have to be reflected in TimeMachine as well, increasing the surface area of this feature by... at least a spike cycle

Reply:
All good! It's a prototype after all.

Wondering if it makes sense to extract this list of `if` statements into a separate file for readability?